Automattic/wp-calypso
# Stepper: implement actual pattern picking in LiB flow

### Changes

This PR implements the entire e2e pattern picking flow:

- Now we have real previews from the block preview API for the patterns. Displayed in iframes. 
- We store the pattern ID in the onboarding store.
- When setting up the site, we get the currently hard-coded pattern content and set it as the homepage content. This applies the pattern successfully. 
- I hard-coded the patterns because the API gives hundreds of patterns with their HTML inlined, terribly slow. I cherry picked the 14 relevant patterns and hard-coded them and mapped them to IDs (this is needed because the preview API still doesn't support previewing by slugs, see: D86662-code ).
- It also replaced the pattern avatar with the uploaded site logo.

### Testing instructions
1. Go to /setup?flow=link-in-bio
2. Play with the pattern picker, test with mobile sizing, etc..
3. Pick a pattern.
4. Make sure to upload an avatar.
5. Finish setting up the site.
6. Go to the site homepage (`YOUR_SITE/home`), it should have the pattern applied and the logo you uploaded.
